packages/flutter_tools/lib/src/build_info.dart

```diff
@@ -137,12 +137,33 @@ enum BuildMode {
   debug,
   profile,
   release,
+
+  // TODO(gspencer): The concept of "dynamic" should be factored out of
```

Comment:
BuildMode is mainly how we decide which engine binary to use `android-arm64-profile` vs `android-arm64-dynamic-profile`, in that regard it makes sense as is